src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmReceiver/ConfirmReceiver.tsx

```diff
@@ -1,25 +1,80 @@
 import React from 'react';
-
 import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import classNames from 'classnames';
+
+import { ACCOUNTS_ENDPOINT } from 'apiCalls';
+import MultiversXIconSimple from 'assets/icons/mvx-icon-simple.svg';
+import { useGetAccountFromApi } from 'hooks';
+import { CopyButton } from 'UI/CopyButton';
+import { ExplorerLink } from 'UI/ExplorerLink';
+import { LoadingDots } from 'UI/LoadingDots';
+import { Trim } from 'UI/Trim';
+import { isContract } from 'utils/smartContracts';
+
+import { ReceiverSubValue } from './components/ReceiverSubValue';
 import styles from './confirmReceiverStyles.scss';
+import { trimReceiverDomain } from './helpers';
 
 export interface ConfirmReceiverPropsType {
-  receiver?: string;
+  receiver: string;
   scamReport: string | null;
 }
 
-export const ConfirmReceiver = (props: ConfirmReceiverPropsType) => {
-  const { receiver, scamReport } = props;
+export const ConfirmReceiver = ({
+  receiver,
+  scamReport
+}: ConfirmReceiverPropsType) => {
+  const isSmartContract = isContract(receiver);
+
+  const {
+    account: usernameAccount,
+    loading: usernameAccountLoading,
+    error: usernameAccountError
+  } = useGetAccountFromApi(isSmartContract ? null : receiver);
+
+  const receiverValue = usernameAccount?.username ?? receiver;
+  const hasUsername = Boolean(
+    receiver && Boolean(usernameAccount?.username) && !usernameAccountError
+  );
 
   return (
     <div className={styles.receiver}>
       <span className={styles.label}>Receiver</span>
 
-      {receiver && (
-        <span className={styles.value} data-testid='confirmReceiver'>
-          {receiver}
+      {receiver && !usernameAccountLoading ? (
```

Comment:
usernameAccountLoading? first:second

check out ternary in ternary (maybe split above in smaller components with Memo)

Reply:
Updated. 👍